src/components/modules/toolbar/conversion.ts

```diff
@@ -1,314 +1,80 @@
-  /**
-   * CSS getter
-   */
-  public static get CSS(): { [key: string]: string } {
-    return {
-      conversionToolbarWrapper: 'ce-conversion-toolbar',
-      conversionToolbarShowed: 'ce-conversion-toolbar--showed',
-      conversionToolbarTools: 'ce-conversion-toolbar__tools',
-      conversionToolbarLabel: 'ce-conversion-toolbar__label',
-      conversionTool: 'ce-conversion-tool',
-      conversionToolHidden: 'ce-conversion-tool--hidden',
-      conversionToolIcon: 'ce-conversion-tool__icon',
-      conversionToolSecondaryLabel: 'ce-conversion-tool__secondary-label',
-
-      conversionToolFocused: 'ce-conversion-tool--focused',
-      conversionToolActive: 'ce-conversion-tool--active',
-    };
-  }
-
-  /**
-   * Conversion Toolbar open/close state
-   *
-   * @type {boolean}
-   */
-  public opened = false;
-
-  /**
-   * Available tools data
-   */
-  private tools: {name: string; toolboxItem: ToolboxConfigEntry; button: HTMLElement}[] = [];
-
-  /**
-   * Instance of class that responses for leafing buttons by arrows/tab
-   *
-   * @type {Flipper|null}
-   */
-  private flipper: Flipper = null;
-
-  /**
-   * Callback that fill be fired on open/close and accepts an opening state
-   */
-  private togglingCallback = null;
-
-  /**
-   * Create UI of Conversion Toolbar
-   */
-  public make(): HTMLElement {
-    this.nodes.wrapper = $.make('div', [
-      ConversionToolbar.CSS.conversionToolbarWrapper,
-      ...(this.isRtl ? [ this.Editor.UI.CSS.editorRtlFix ] : []),
-    ]);
-    this.nodes.tools = $.make('div', ConversionToolbar.CSS.conversionToolbarTools);
-
-    const label = $.make('div', ConversionToolbar.CSS.conversionToolbarLabel, {
-      textContent: I18n.ui(I18nInternalNS.ui.inlineToolbar.converter, 'Convert to'),
-    });
-
-    /**
-     * Add Tools that has 'import' method
-     */
-    this.addTools();
-
-    /**
-     * Prepare Flipper to be able to leaf tools by arrows/tab
-     */
-    this.enableFlipper();
-
-    $.append(this.nodes.wrapper, label);
-    $.append(this.nodes.wrapper, this.nodes.tools);
-
-    return this.nodes.wrapper;
-  }
-
-  /**
-   * Deactivates flipper and removes all nodes
-   */
-  public destroy(): void {
-    /**
-     * Sometimes (in read-only mode) there is no Flipper
-     */
-    if (this.flipper) {
-      this.flipper.deactivate();
-      this.flipper = null;
-    }
-
-    this.removeAllNodes();
-  }
-
+export default class Conversion extends Module {
   /**
-   * Toggle conversion dropdown visibility
+   * Returns list of all available conversion menu items for specified block
    *
-   * @param {Function} [togglingCallback] — callback that will accept opening state
+   * @param block - block to get conversion items for
    */
-  public toggle(togglingCallback?: (openedState: boolean) => void): void {
-    if (!this.opened) {
-      this.open();
-    } else {
-      this.close();
-    }
+  public async getItemsForBlock(block: Block): Promise<PopoverItemDefaultParams[]> {
```

Comment:
lets create utility `getConvertibleToolsForBlock` which will accept all tools and a block and return a ToolsCollection.
Then you will map that collection to popover item.

Reply:
done